protocol_driver_double_barrel.cc

```diff
@@ -94,28 +91,53 @@ absl::Status ProtocolDriverDoubleBarrel::HandleConnect(
   return barrel_2_->HandleConnect(remote_connection_info, peer);
 }
 
+#define APPEND_VECTOR(a, b) \
+    (a.insert(a.end(), b.begin(), b.end()))
+
 void ProtocolDriverDoubleBarrel::HandleConnectFailure(
     std::string_view local_connection_info) {
   barrel_1_->HandleConnectFailure(local_connection_info);
   barrel_2_->HandleConnectFailure(local_connection_info);
 }
 
 std::vector<TransportStat> ProtocolDriverDoubleBarrel::GetTransportStats() {
-  std::vector<TransportStat> stats;
-  return stats;
+  std::vector<TransportStat> transport_stats;
+
+  transport_stats.push_back({"barrel_1_server_stats", {0}});
+  auto barrel_1_server_stats = ((ProtocolDriverServer*)
+                                barrel_1_.get())->GetTransportStats();
+  APPEND_VECTOR(transport_stats, barrel_1_server_stats);
+
+  transport_stats.push_back({"barrel_1_client_stats", {0}});
+  auto barrel_1_client_stats = ((ProtocolDriverClient*)
+                                barrel_1_.get())->GetTransportStats();
+  APPEND_VECTOR(transport_stats, barrel_1_client_stats);
+
+  transport_stats.push_back({"barrel_2_server_stats", {0}});
+  auto barrel_2_server_stats = ((ProtocolDriverServer*)
+                                barrel_2_.get())->GetTransportStats();
+  APPEND_VECTOR(transport_stats, barrel_2_server_stats);
+
+  transport_stats.push_back({"barrel_2_client_stats", {0}});
+  auto barrel_2_client_stats = ((ProtocolDriverClient*)
+                                barrel_2_.get())->GetTransportStats();
+  APPEND_VECTOR(transport_stats, barrel_2_client_stats);
+
+  return transport_stats;
 }
 
 void ProtocolDriverDoubleBarrel::InitiateRpc(int peer_index, ClientRpcState* state,
                                      std::function<void(void)> done_callback) {
-  if (use_barrel_1_) {
+  if (std::atomic_fetch_xor_explicit(&use_barrel_1_, 1, std::memory_order_relaxed)) {
```

Comment:
0x1 & atomic_fetch_add_explicit(...)
atomic_fetch_xor is not available universally

Reply:
fixed